### PR TITLE
pdnssec: Check if addresses exist for TLSA hosts

### LINKED ISSUE #2432: Make check-zone warn about TLSA record preventing wildcard match for www record
We managed to make our website unavailable by simply adding a DANE record to our DNS. This made NSEC3 prove non-existence of our www record, which made everyone using DNSSEC unable to use our website. I believe this is a bug inside PowerDNS, I haven't found any reason why this behavior should happen. We are running PowerDNS 3.4.3 with MySQL backend and Poweradmin interface.

Before the problem occurred we had basic records in our DNS. We did not have a 'www' record, but instead we used wildcard entry (*.example.net) pointing to our webservers IP. The zone used NSEC3 and everything was working fine.

After entering a TLSA record to the zone (_443._tcp.www.example.net TLSA <dane info>), NSEC3 was suddenly proving non-existence of www.example.net. Making our website unavailable for everyone using DNSSEC or the Google DNS. Using 'pdnssec rectify-zone' did not help at all, and 'pdnssec check-zone' did not report any errors. 

Adding a www.example.net entry did fix this problem eventually. I can't find any reason why this should be required as the wildcard already points to the same address. 

The bug was successfully reproduced on another domain on another server as well.

### PR DESCRIPTION
**Pull request by pieterlexis**
When a record `_443._tcp.www.example.com.|TLSA` exists and no record for
`www.example.com.|AAAA` or `A`, we emit a warning. Similarly, when
`*.example.com.|AAAA` or `A` _does_ exist, we emit a warning, as
`www.example.com.` becomes an empty non-terminal at that point.

This is WIP, as we don't check for CNAMEs (and we should follow them in the same zone). Furthermore, I'm not totally happy with 2 more sets with records, so a review is needed.

This should close #2432 once merged.

### CODE REVIEW

**Comment by mind04** [comment]
Right now you are checking for a very specific error case. The check should be more general: is there a address record or a cname for host covered by the tlsa record (yes/no). 
If not warn.

Flowing cnames is something for an other pull if you ask me.

**Comment by pieterlexis** [comment]
@mind04: agreed on the A||AAAA||CNAME part. I still feel that a separate warning for a wildcard is in order (with perhaps a better explanation) as one who is not extremely well-versed in DNS name semantics can understand (or know) why a wildcard won't match.

**Comment by mind04** [comment]
You can't fix a lack of dns knowledge with a warning, The right place for education is documentation or a nice blog post about wildcards in dns.

**Comment by pieterlexis** [comment]
needs review

File: pdns/pdnssec.cc [hunk 1]
```diff
@@ -613,6 +615,26 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone)
     }
   }
 
+  for(const auto &i: tlsas) {
+    DNSName name = DNSName(i);
+    name.trimToLabels(name.getRawLabels().size()-2);
+    if (cnames.find(name) == cnames.end() && noncnames.find(name) == noncnames.end()) {
+      // No specific record for the name in the TLSA record exists, this
+      // is already worth emitting a warning. Let's see if a wildcard exitst.
```
Comment (Habbie) [inline pdns/pdnssec.cc]: exitst

**Comment by Habbie** [comment]
one nit, then LGTM

**Pull request merged**